allennlp/models/semantic_parsing/atis/atis_semantic_parser.py

```diff
@@ -0,0 +1,592 @@
+                probabilities = action_debug_info['probabilities']
+                actions = []
+                for action, probability in zip(considered_actions, probabilities):
+                    if action != -1:
+                        actions.append((action_mapping[(batch_index, action)], probability))
+                actions.sort()
+                considered_actions, probabilities = zip(*actions)
+                action_info['considered_actions'] = considered_actions
+                action_info['action_probabilities'] = probabilities
+                action_info['utterance_attention'] = action_debug_info.get('question_attention', [])
+                instance_action_info.append(action_info)
+            batch_action_info.append(instance_action_info)
+        output_dict["predicted_actions"] = batch_action_info
+        return output_dict
+
+    @overrides
+    def forward(self,  # type: ignore
+                utterance: Dict[str, torch.LongTensor],
+                world: List[AtisWorld],
+                actions: List[List[ProductionRuleArray]],
+                linking_scores: torch.Tensor,
+                target_action_sequence: torch.LongTensor = None,
+                sql_queries: List[str] = None) -> Dict[str, torch.Tensor]:
+        # pylint: disable=arguments-differ
+        """
+        We set up the initial state for the decoder, and pass that state off to either a DecoderTrainer,
+        if we're training, or a BeamSearch for inference, if we're not.
+
+        Parameters
+        ----------
+        utterance : Dict[str, torch.LongTensor]
+            The output of ``TextField.as_array()`` applied on the utterance ``TextField``. This will
+            be passed through a ``TextFieldEmbedder`` and then through an encoder.
+        world : ``List[AtisWorld]``
+            We use a ``MetadataField`` to get the ``World`` for each input instance.  Because of
+            how ``MetadataField`` works, this gets passed to us as a ``List[AtisWorld]``,
+        actions : ``List[List[ProductionRuleArray]]``
+            A list of all possible actions for each ``World`` in the batch, indexed into a
+            ``ProductionRuleArray`` using a ``ProductionRuleField``.  We will embed all of these
+            and use the embeddings to determine which action to take at each timestep in the
+            decoder.
+        linking_scores: ``torch.Tensor``
+            A matrix of the linking the utterance tokens and the entities. This is a binary matrix that
+            is deterministically generated where each entry indicates whether a token generated an entity.
+            This tensor has shape ``(num_entities, num_utterance_tokens)``.
+        target_action_sequences : torch.Tensor, optional (default=None)
+            A list of possibly valid action sequences, where each action is an index into the list
+            of possible actions.  This tensor has shape ``(batch_size, num_action_sequences,
```

Comment:
`num_action_sequences` is always 1 for you.  If you have this inside a `ListField` in your dataset reader, I'd probably remove that, and just have inputs of size `(batch_size, sequence_length)`.  You can `.unsqueeze()` it when passing it to `MaximumMarginalLikelihood` (and @rajasagashe is planning on submitting a `MaximumLikelihood` class sometime soon, which we can use instead of `MaximumMarginalLikelihood` once it's in).